### PR TITLE
Add dialog-box warning support 

### PR DESCRIPTION
**Pull request by spacegaier**
## Proposed change

Helper function to create a question dialog with default Yes/No option (no button selection so user has to purposely select an option). The confirm and dismiss texts can still be override as before. 
Also added the option to flag a dialog as containing a warning so that the matching CSS class gets applied.

The idea came to me when I changed the name of a device. The user then gets asked whether the entity IDs should also be changed. I saw two issues here:
1. It was a Yes / Cancel dialog, but you cannot actually cancel anything here (the device renaming already happened, so Yes/No is correct IMHO).
2. Clicking Yes here has a comparably big impact and to make sure a user is aware of the potential ramifications, this should be visually highlighted.

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/dialogs/generic/show-dialog-box.ts [hunk 1]
```diff
@@ -76,12 +79,21 @@ export const showAlertDialog = (
...
   showDialogHelper(element, dialogParams, { confirmation: true }) as Promise<
     boolean
   >;
 
+export const showQuestionDialog = (
+  element: HTMLElement,
+  dialogParams: DialogParams
+) =>
+  showDialogHelper(element, dialogParams, {
+    question: true,
```
Comment (bramkragten) [inline src/dialogs/generic/show-dialog-box.ts:93]: Instead of this extra property, can't we just pass the `confirmText` and `dismissText`?

Comment (spacegaier) [reply src/dialogs/generic/show-dialog-box.ts:93]: Yes, I think we can. Originally there was additional logic inside based on this property, but that was in the end not needed.

Comment (spacegaier) [reply src/dialogs/generic/show-dialog-box.ts:93]: Currently we do not have `hass` available here to call `localize()`. Is there an easy way to get access to it here? I think that was the reason why I went with my approach instead.

Comment (bramkragten) [reply src/dialogs/generic/show-dialog-box.ts:93]: Add `hass` (or just lokalize) as a argument of `showQuestionDialog`?

**Review by bramkragten — APPROVED** [review]

**Pull request merged**